### ISSUE TITLE
[gazebo_plugins] libgazebo_ros_camera - some objects incorrectly occluded

### ISSUE THREAD

**Issue by xperroni** [ros-simulation/gazebo_ros_pkgs#417]
Camera images published by `libgazebo_ros_camera` show certain objects incorrectly occluded by others, or even by the ground plane itself.

**Steps to reproduce**
1. Start `gazebo_ros` with the attached world file;
2. Open RQT and connect the image viewer to the `/camera/image_raw` topic;
3. On the Gazebo UI, check option _View->Transparent_;
4. Check the RQT image viewer again.

See the attached ZIP archive for the world file and complementary files.

[occlusion_bug_files.zip](https://github.com/ros-simulation/gazebo_ros_pkgs/files/201979/occlusion_bug_files.zip)

**Results**
- **Expected:** the image on RQT should show the FIRST 2015 trash can model in full, as it's standing in front of the camera;
- **Actual:** Only the top half of the trash can is visible. When switching the _Transparent_ option on Gazebo, it can be seen on RQT that the remaining of the trash can is standing "behind" the ground plane, which was erroneously rendered over it.

See attached images for actual results from both Gazebo and RQT.

I'm using Gazebo 7 with ROS Indigo on Kubuntu 14.04.1.

View of the scene from Gazebo:

![view_from_gazebo](https://cloud.githubusercontent.com/assets/1932656/14241752/23b1ec1a-fa88-11e5-90ee-2d865ad78684.png)

Camera image preview as seen from inside Gazebo:

![camera_image_from_gazebo](https://cloud.githubusercontent.com/assets/1932656/14241749/2380f560-fa88-11e5-8b26-8793a66496ad.png)

Camera image from RQT:

![camera_image_from_rqt](https://cloud.githubusercontent.com/assets/1932656/14241750/238d95a4-fa88-11e5-8f0c-15b0c4d86628.png)

Camera image from RQT after the _Transparent_ option is selected on Gazebo:

![camera_image_from_rqt_transparent](https://cloud.githubusercontent.com/assets/1932656/14241751/2393d5ae-fa88-11e5-851c-4abf9abd9645.png)

**Comment by skohlbr** [comment]
I believe the same issue is tracked on the Gazebo Bitbucket issue tracker here: https://bitbucket.org/osrf/gazebo/issues/1837/vmware-rendering-z-ordering-appears-random

**Comment by xperroni** [comment]
Yes, it looks very much like my problem. Unfortunately it seems the only way to fix it is to recompile Gazebo with anti-aliasing disabled, though.

**Comment by j-rivero** [comment]
Let's keep track of the problem in the [gazebo issue](https://bitbucket.org/osrf/gazebo/issues/1837/vmware-rendering-z-ordering-appears-random)
